pkg/skaffold/deploy/helm/deploy.go

```diff
@@ -231,7 +235,7 @@ func (h *Deployer) Dependencies() ([]string, error) {
 // Cleanup deletes what was deployed by calling Deploy.
 func (h *Deployer) Cleanup(ctx context.Context, out io.Writer) error {
 	for _, r := range h.Releases {
-		releaseName, err := util.ExpandEnvTemplate(r.Name, nil)
+		releaseName, err := util.ExpandEnvTemplateOrFail(r.Name, nil)
```

Comment:
I wasnt aware release names supports env templates. 
Probably should be added to docs  here https://skaffold.dev/docs/references/yaml/#deploy-helm-releases-name